handlers/utils.go

```diff
@@ -22,13 +22,59 @@ const defaultPatchType = "merge"
 
 var defaultPromClientSupplier = prometheus.NewClient
 
-func checkNamespaceAccess(ctx context.Context, nsServ business.NamespaceService, namespace string) (*models.Namespace, error) {
-	return nsServ.GetNamespace(ctx, namespace)
+func checkNamespaceAccess(ctx context.Context, nsServ business.NamespaceService, namespace string, cluster string) (*models.Namespace, error) {
+	return nsServ.GetNamespaceByCluster(ctx, namespace, cluster)
 }
 
-func createMetricsServiceForNamespace(w http.ResponseWriter, r *http.Request, promSupplier promClientSupplier, namespace string) (*business.MetricsService, *models.Namespace) {
-	metrics, infoMap := createMetricsServiceForNamespaces(w, r, promSupplier, []string{namespace})
-	if result, ok := infoMap[namespace]; ok {
+// Create Metrics Service for a single namespace and no cluster parameter. It will check permissions for all the clusters, and return the service
+// if it has permissions at least one of the clusters
+func createMetricsServiceForNamespaceMC(w http.ResponseWriter, r *http.Request, promSupplier promClientSupplier, ns models.Namespace) (*business.MetricsService, *models.Namespace) {
+	layer, err := getBusiness(r)
+	if err != nil {
+		RespondWithError(w, http.StatusInternalServerError, err.Error())
+		return nil, nil
+	}
+	prom, err := promSupplier()
+	if err != nil {
+		log.Error(err)
+		RespondWithError(w, http.StatusServiceUnavailable, "Prometheus client error: "+err.Error())
+		return nil, nil
+	}
+	var nsInfo nsInfoError
+
+	nsLists, err := layer.Namespace.GetNamespaceClusters(r.Context(), ns.Name)
+	if err != nil || len(nsLists) == 0 {
+		log.Infof("No clusters found.")
+		// TWe cant find the cluster list, we asume it is local
```

Comment:
Shouldn't the local cluster be included in `GetNamespaceClusters`? `err` could be nil for a variety of reasons so I'm not sure we can assume this?

Reply:
Well, the local cluster should be included in `GetNamespaceCluster`. I found this problem in the tests, so I figured out that it could be any scenario when it is not possible to retrieve these list so we could make this assumption?

Reply:
I will ping @jshaughn in case he has some ideas here

Reply:
I just found this issue in testing scenario. I added a TODO to review further.